app/src/main/java/com/bakkenbaeck/sol/BaseApplication.kt

```diff
@@ -0,0 +1,133 @@
+package com.bakkenbaeck.sol
+
+import android.app.AlarmManager
+import android.app.Application
+import android.app.PendingIntent
+import android.content.Intent
+import android.graphics.BitmapFactory
+import android.support.v4.app.NotificationCompat
+import android.text.Html
+import android.util.Log
+import com.bakkenbaeck.sol.extension.getAlarmService
+import com.bakkenbaeck.sol.extension.getNotificationService
+import com.bakkenbaeck.sol.repository.LocationRepository
+import com.bakkenbaeck.sol.service.AlarmReceiver
+import com.bakkenbaeck.sol.util.SolPreferences
+import com.bakkenbaeck.sol.view.SunActivity
+
+class BaseApplication : Application() {
+
+    companion object {
+        const val ACTION_UPDATE = "com.bakkenbaeck.sol.UPDATE"
+        const val EXTRA_DAILY_MESSAGE = "daily_message"
+        const val EXTRA_SUNRISE_TIME = "sunrise_time"
+        const val EXTRA_SUNSET_TIME = "sunset_time"
+        const val EXTRA_LOCATION_MESSAGE = "location_message"
+        const val EXTRA_CURRENT_PHASE = "current_phase"
+
+        lateinit var instance: BaseApplication
+    }
+
+    lateinit var locationRepository: LocationRepository
+
+    init {
+        instance = this
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        init()
+    }
+
+    private fun init() {
+        initLocationRepository()
+        initCalligraphy()
+        initObservers()
+    }
+
+    private fun initLocationRepository() {
+        locationRepository = LocationRepository(applicationContext)
+    }
+
+    private fun initCalligraphy() {
+        try {
+//            CalligraphyConfig.initDefault(CalligraphyConfig.Builder()
+//                    .setDefaultFontPath("fonts/gtamericalight.ttf")
+//                    .setFontAttrId(R.attr.fontPath)
+//                    .build()
+//            )
+        } catch (e: Exception) {
+            Log.d("BaseApplication", "Could not add font")
+        }
+    }
+
+    private fun initObservers() {
+        locationRepository.notification.observeForever {
```

Comment:
Use `Transformations` (https://developer.android.com/reference/android/arch/lifecycle/Transformations) but to be honest, having so much code running in the Application seems incorrect. Does it have to be this way?

Reply:
Incorrect according to who/what? Are you saying it is incorrect because it has a larger than necessary memory-footprint?

I just pushed a new commit where I changed the repository-class back to being a service (`IntentService` this time actually) and moved out the `FusedLocationProviderClient` to the `BaseApplication`-class so it is initialized once instead of every time the `IntentService` runs.

Reply:
Incorrect because an Application is the entry point of execution for the Android OS. The Application class should be used for Application-wide logic (perhaps initialising dependency injection, a crash framework, a custom font) but not containing logic.

Say for example you wanted to test this logic, it'd be easier in a pure Kotlin class than in the Application class.